lib/gst/player/gstplayer.c

```diff
@@ -2641,6 +2658,46 @@ gst_player_seek_internal (gpointer user_data)
 }
 
 /**
+ * gst_player_set_rate:
+ * @player: #GstPlayer instance
+ * @rate: playback rate
+ *
+ * Playback at specified rate
+ */
+void
+gst_player_set_rate (GstPlayer *self, gdouble rate)
+{
+  g_return_if_fail (GST_IS_PLAYER (self));
+
+  self->rate = rate;
+  if (rate != 1.0)
+    self->trick_play = TRUE;
+  else
+    self->trick_play = FALSE;
```

Comment:
Why not just check for self->rate != 1.0 everywhere instead of having this boolean?